### PR TITLE
fix time in raster stream xarray conversion

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Raster Stream Notebook kann den Plot in der letzten Zelle nicht erzeugen.](https://trello.com/c/QK54ZxKk/1295-raster-stream-notebook-kann-den-plot-in-der-letzten-zelle-nicht-erzeugen)

File: geoengine/workflow.py [hunk 1]
```diff
@@ -742,6 +742,9 @@ def merge_tiles(tiles: List[xr.DataArray]) -> Optional[xr.DataArray]:
             )
         )
 
+        # Ensure the 'time' coordinate has a datetime64 data type
+        output['time'] = output['time'].astype('datetime64[ns]')
+
```
Comment (ChristianBeilschmidt) [inline geoengine/workflow.py:747]: update xarray dependecy to 2023.x instead?

https://github.com/pydata/xarray/issues/6416

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: geoengine/workflow.py (hunk 1 above)
Comment (ChristianBeilschmidt) [reply geoengine/workflow.py:747]: Do we have datetime64[ns] before merging?

We do `time = TimeInterval.from_response(json.loads(metadata[b'time']))` for a single raster.
Is this always datetime64[ns]? Shouldn't this be [ms] anyway :thinking: .

Comment (michaelmattig) [reply geoengine/workflow.py:747]: this is the output:
```
# array 0
 <xarray.DataArray 'time' ()>
array(Timestamp('2014-04-01 02:00:00+0000', tz='UTC'), dtype=object)
Coordinates:
    time         object 2014-04-01T02:00:00+00:00
    spatial_ref  int64 0

# array 1
<xarray.DataArray 'time' ()>
array(Timestamp('2014-05-01 02:00:00+0000', tz='UTC'), dtype=object)
Coordinates:
    time         object 2014-05-01T02:00:00+00:00
    spatial_ref  int64 0

# array 2
<xarray.DataArray 'time' ()>
array(Timestamp('2014-06-01 02:00:00+0000', tz='UTC'), dtype=object)
Coordinates:
    time         object 2014-06-01T02:00:00+00:00
    spatial_ref  int64 0

# merged
<xarray.DataArray 'time' (time: 3)>
array([1396317600000000000, 1398909600000000000, 1401588000000000000],
      dtype=object)
Coordinates:
  * time         (time) object 1396317600000000000 ... 1401588000000000000
    spatial_ref  int64 0

# merged with astype
<xarray.DataArray 'time' (time: 3)>
array(['2014-04-01T02:00:00.000000000', '2014-05-01T02:00:00.000000000',
       '2014-06-01T02:00:00.000000000'], dtype='datetime64[ns]')
Coordinates:
  * time         (time) datetime64[ns] 2014-04-01T02:00:00 ... 2014-06-01T02:...
    spatial_ref  int64 0
```

Comment (ChristianBeilschmidt) [reply geoengine/workflow.py:747]: So maybe the "solution" is to set it "correctly" as datetime64 right from the start and then see if it works then.
Anyway, it should also be of the correct type for a single time step xarray.

And if it still does not work then after concat, we need to use astype.

Comment (michaelmattig) [reply geoengine/workflow.py]: I set the type for the individual xarrays now instead of after the marge

549162ef2fce2d93dce9bc5cb166c0ec44bdbce0

Comment (jdroenner) [reply geoengine/workflow.py]: in workflow.py it iit says:

` 'time': np.datetime64(time.start, 'ms'),  # TODO: incorporate time end?" `

in the example the type is `datetime64[ns] ` is that correct?

Comment (michaelmattig) [reply geoengine/workflow.py]: your are right, however, the doc says https://docs.xarray.dev/en/stable/user-guide/time-series.html

> Creating datetime64 data
Xarray uses the numpy dtypes datetime64[ns] and timedelta64[ns] to represent datetime data, which offer vectorized (if sometimes buggy) operations with numpy and smooth integration with pandas.

so I think we cannot do anything about this.

Comment (jdroenner) [reply geoengine/workflow.py]: so it always converts to "ns"?

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**